### PR TITLE
[docs] add xpu device check 

### PR DESCRIPTION
**Pull request by faaany**
## What does this PR do?
There are many places in the task guides that only check the cuda device availability. This PR adds xpu besides cuda, but I am not sure whether my current change is the best one, because there are other ways to add xpu as, e.g. 
- add xpu in the inline-comment 
- use accelerate's `get_backend` API to get the underlying device 

Just let me what is your favorite, @stevhliu Once agreed by you, I can submit further commits to replicate the change in other places.

### CODE REVIEW

**Review by stevhliu — COMMENTED** [review]
Thanks for the suggestion!

I think using `get_backend` may be the easiest option for users, and this will also include other devices such as `mps`.

**Comment by faaany** [comment]
> Thanks for the suggestion!
> 
> I think using `get_backend` may be the easiest option for users, and this will also include other devices such as `mps`.

thanks! code update. Is this okay for you? Then I can change to other places as well.

File: docs/source/en/tasks/semantic_segmentation.md [hunk 1]
```diff
@@ -689,7 +689,8 @@ Reload the dataset and load an image for inference.
 We will now see how to infer without a pipeline. Process the image with an image processor and place the `pixel_values` on a GPU:
 
 ```py
->>> device = torch.device("cuda" if torch.cuda.is_available() else "cpu")  # use GPU if available, otherwise use a CPU
+>>> from accelerate.test_utils.testing import get_backend
+>>> device, _, _ = get_backend()
```
Comment (stevhliu) [inline docs/source/en/tasks/semantic_segmentation.md:693]: Maybe add a comment here that explains what's happening. Perhaps something like:

```suggestion
# automatically detects the underlying device type (GPU, CUDA, CPU, XPU, MPS, etc.)
>>> device, _, _ = get_backend()
```

**Review by stevhliu — COMMENTED** [review]
Very nice, thank you! 👍

**Comment by faaany** [comment]
> Very nice, thank you! 👍

I only made one change: I removed GPU from the comment because GPU is a general term including CUDA, XPU, NPU. And I made the same change at other places as well. Pls have a review, thx a lot!

**Review by stevhliu — APPROVED** [review]
Fantastic, works for me!

**Pull request merged**